Issues importing data from a relative path

I've been experimenting with integrating this package into a project of mine and I've run into an issue. Here's the [git repo](https://tulpa.dev/cadey/mi/src/branch/dhall). I tried to use the following Go code:

```go
package main

import (
	"io/ioutil"

	"github.com/kr/pretty"
	"github.com/philandstuff/dhall-golang"
)

type Config struct {
	Discord struct {
		Webhook string `json:"Webhook"`
	} `json:"Discord"`
	Gitea struct {
		Instance string `json:"Instance"`
		Token    string `json:"Token"`
	} `json:"Gitea"`
	Github struct {
		Token string `json:"Token"`
	} `json:"Github"`
	Mastodon struct {
		Account   string `json:"Account"`
		AppID     string `json:"AppID"`
		AppSecret string `json:"AppSecret"`
		Instance  string `json:"Instance"`
		Token     string `json:"Token"`
	} `json:"Mastodon"`
	Paseto struct {
		Private string `json:"Private"`
		Public  string `json:"Public"`
	} `json:"Paseto"`
	Patreon struct {
		AccessToken  string `json:"AccessToken"`
		ClientID     string `json:"ClientID"`
		ClientSecret string `json:"ClientSecret"`
		RefreshToken string `json:"RefreshToken"`
	} `json:"Patreon"`
	Reddit struct {
		AppID     string `json:"AppID"`
		AppSecret string `json:"AppSecret"`
		Password  string `json:"Password"`
		Username  string `json:"Username"`
	} `json:"Reddit"`
	RethinkDBURL string `json:"RethinkDBURL"`
	Twitter      struct {
		APIKey         string `json:"ApiKey"`
		APISecret      string `json:"ApiSecret"`
		ConsumerSecret string `json:"ConsumerSecret"`
		ConsumerToken  string `json:"ConsumerToken"`
	} `json:"Twitter"`
}

func main() {
	var cfg Config
	data, err := ioutil.ReadFile("mi.dhall")
	if err != nil {
		panic(err)
	}
	err = dhall.Unmarshal([]byte(data), &cfg)
	if err != nil {
		panic(err)
	}
	pretty.Println(cfg)
}
```

to import [this folder of dhall files](https://tulpa.dev/cadey/mi/src/branch/dhall/cfg) via this expression:

```dhall
let mi = ./cfg/package.dhall in mi.Mi::{=}
```

but I get this error:

```console
$ go run ./cmd/mi-cfg/
panic: unknown term type./cfg/package.dhall

goroutine 1 [running]:
github.com/philandstuff/dhall-golang/core.evalWith(0x7f6224428110, 0xc000116cc0, 0xc00039a7e0, 0x0, 0x0, 0x1)
        /home/cadey/go/pkg/mod/github.com/philandstuff/dhall-golang@v0.0.0-20200128213416-5ebf1681a323/core/eval.go:604 +0x7610
github.com/philandstuff/dhall-golang/core.evalWith(0x83eca0, 0xc00039a720, 0xc00039a7b0, 0x83ec00, 0xc00039a720, 0x1)
        /home/cadey/go/pkg/mod/github.com/philandstuff/dhall-golang@v0.0.0-20200128213416-5ebf1681a323/core/eval.go:128 +0x722a
github.com/philandstuff/dhall-golang/core.Eval(...)
        /home/cadey/go/pkg/mod/github.com/philandstuff/dhall-golang@v0.0.0-20200128213416-5ebf1681a323/core/eval.go:14
github.com/philandstuff/dhall-golang.Unmarshal(0xc0000f2000, 0x7b6, 0x9b6, 0x72f740, 0xc0000b6180, 0x0, 0x0)
        /home/cadey/go/pkg/mod/github.com/philandstuff/dhall-golang@v0.0.0-20200128213416-5ebf1681a323/unmarshal.go:34 +0xef
main.main()
        /home/cadey/code/mi/cmd/mi-cfg/main.go:59 +0xbe
exit status 2
```

I believe I am using the most recent version of the dhall-golang package [based on this commit](https://github.com/philandstuff/dhall-golang/commit/5ebf1681a323abc5692f5eca047ce8886b4933e3). Am I doing anything wrong or have I found a bug?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,9 +9,10 @@
    - removed `core.AppValue` from public interface
    - renamed various types to remove `-Val` and `-Term` suffixes
 
-### Changed
+### Fixed
 
- - `dhall.Unmarshal()` now typechecks before evaluating
+ - `dhall.Unmarshal()` now resolves imports and typechecks before
+   evaluating
 
 ### Added
 

--- a/testdata/unmarshal-test.dhall
+++ b/testdata/unmarshal-test.dhall
@@ -1,0 +1,1 @@
+{ Port = 5050, Name = "inetd" }

--- a/unmarshal.go
+++ b/unmarshal.go
@@ -6,6 +6,7 @@ import (
 	"strings"
 
 	"github.com/philandstuff/dhall-golang/core"
+	"github.com/philandstuff/dhall-golang/imports"
 	"github.com/philandstuff/dhall-golang/parser"
 )
 
@@ -18,18 +19,23 @@ func isMapEntryType(recordType core.RecordType) bool {
 	return false
 }
 
-// Unmarshal takes dhall input as a byte array and parses it,
-// evaluates it, and unmarshals it into the given variable.
+// Unmarshal takes dhall input as a byte array and parses it, resolves
+// imports, typechecks, evaluates, and unmarshals it into the given
+// variable.
 func Unmarshal(b []byte, out interface{}) error {
 	term, err := parser.Parse("-", b)
 	if err != nil {
 		return err
 	}
-	_, err = core.TypeOf(term)
+	resolved, err := imports.Load(term)
 	if err != nil {
 		return err
 	}
-	Decode(core.Eval(term), out)
+	_, err = core.TypeOf(resolved)
+	if err != nil {
+		return err
+	}
+	Decode(core.Eval(resolved), out)
 	return nil
 }
 

--- a/unmarshal_test.go
+++ b/unmarshal_test.go
@@ -202,12 +202,23 @@ var _ = Describe("Decode", func() {
 var _ = Describe("Unmarshal", func() {
 	It("Parses 1 + 2", func() {
 		var actual uint
-		Unmarshal([]byte("1 + 2"), &actual)
+		err := Unmarshal([]byte("1 + 2"), &actual)
+		Expect(err).ToNot(HaveOccurred())
 		Expect(actual).To(Equal(uint(3)))
 	})
 	It("Throws a type error for `1 + -2`", func() {
 		var actual uint
 		err := Unmarshal([]byte("1 + -2"), &actual)
 		Expect(err).To(HaveOccurred())
+	})
+	It("Fetches imports", func() {
+		type Config struct {
+			Port int
+			Name string
+		}
+		var actual Config
+		err := Unmarshal([]byte("./testdata/unmarshal-test.dhall"), &actual)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(actual).To(Equal(Config{Port: 5050, Name: "inetd"}))
 	})
 })

